sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexClientBuilder.java

```diff
@@ -57,16 +109,54 @@
  * </pre>
  * <!-- end com.azure.search.documents.indexes.SearchIndexAsyncClient.instantiation -->
  *
- * <p><strong>Instantiating a synchronous Search Index Client</strong></p>
  *
- * <!-- src_embed com.azure.search.documents.indexes.SearchIndexClient.instantiation -->
+ * <h4>
+ *     Building and Authenticating a {@link SearchIndexClient} or {@link SearchIndexAsyncClient} using Azure Active Directory
+ * </h4>
+ *
+ * <p>
+ *   You can also create a {@link SearchIndexClient} or {@link SearchIndexAsyncClient} using Azure Active Directory (AAD)
+ *   authentication. Your user or service principal must be assigned the "Search Index Data Reader" role. Using the
```

Comment:
Doesn't the required role change based on whether the clients will be used for "read" operations (list, get) vs "write" operations (create, delete)?